New api for DefinePlugin (#3916)
webpack/webpack.js.org: src/content/plugins/define-plugin.md

  'SERVICE_URL': JSON.stringify('https://dev.example.com')
});
```

### Rebuild 

> Rebuild wehen the schema updates.

Reviewer: ```suggestion
Rebuild when the schema updates.
```

Reviewer 2: I think Schema was only mentioned in the PR as an example... This feature allows you to actually watch for changes in runtimeValue or am i missing something? https://github.com/webpack/webpack/pull/6793/files

Author: I think you are correct

```javascript
new webpack.DefinePlugin({

>   BUILT_AT: webpack.DefinePlugin.runtimeValue(Date.now, [fileDep, fileDep])

Reviewer 2: This example should be extended to explain fileDep, e.g. require a file and pass it as an example dependency of the runtime value. Then demonstrate what happens when file dep changes

Reviewer 2: runtimeValue is new to this plugin, its missing its heading and props explanation/types

Author: Done

})
```

> T> Passing `true` as second argument instead of `fileDependencies` will flag the module as uncachable.

Reviewer: ```suggestion
T> Passing `true` as second argument instead of `fileDependencies` will flag the module as uncacheable.
```

Author: thanks :)

  'SERVICE_URL': JSON.stringify('https://dev.example.com')
});
```

### Rebuild 

Rebuild where there are changes in runtimeValue.

__webpack.app.config.js__

```javascript

const fileDep = path.resolve(__dirname, 'sample.txt');

> new webpack.DefinePlugin({
>   BUILT_AT: webpack.DefinePlugin.runtimeValue(Date.now, [fileDep]);
> })

Reviewer 2: ```suggestion
new webpack.DefinePlugin({
  BUILT_AT: webpack.DefinePlugin.runtimeValue(Date.now, [fileDep])
});
```

```

> The first argument to `runtimeValue` is a `function` that returns the value to be assigned and the second argument is array of `fileDependencies` (what files to be watched for). 

Reviewer 2: ```suggestion
The first argument to `runtimeValue` is a `function` that returns the value to be assigned and the second argument is an array of `fileDependencies` (what files to be watch for). 
```

> In the previous example The value of `BUILD_AT` would be the time at which the fileDep was last updated. So if the the file was updated at 1597778473060 then BUILD_AT would have 1597778473060 as it value.

Reviewer 2: ```suggestion
In the previous example, the value of `BUILD_AT` would be the time at which the fileDep was last updated. If the the file was updated at `1597778473060` then `BUILD_AT` would have `1597778473060` as its value.
```

> In the previous example, the value of `BUILD_AT` would be the time at which the fileDep was last updated. If the the file was updated at `1597778473060` then `BUILD_AT` would have `1597778473060` as its value.

Reviewer 2: ```suggestion
In the previous example, the value of `BUILD_AT` would be the time at which the file passed as `fileDep` was last updated in the file system.
```
